Tests/reviews.api.test.js

```diff
@@ -0,0 +1,133 @@
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviews/${invalidProductId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(404);
+        expect(Array.isArray(response.body)).toEqual(true);
+        var insertedData = response.body;
+        expect(insertedData[0]).toBeUndefined();
+      });
+
+  });
+
+
+  test("/Reviews/getReviewSummary/:productId : it should respond with an object, respond 200 for valid response ", async () => {
+    var productId = 1000;
+    var mockData = await SeedData.generateSeedData(10, 1);//load 10 customer,  for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviewSummary/${productId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toEqual(false);
+        expect(typeof (response.body)).toEqual('object');
+        var insertedData = response.body;
+        expect.objectContaining({
+          averageRating: expect.any(String),
+          totalRatings: expect.any(Number),
+          fiveStar: expect.any(String),
+          fourStar: expect.any(String),
+          threeStar: expect.any(String),
+          twoStar: expect.any(String),
+          oneStar: expect.any(String)
+        });
+      });
+
+  });
+
+  test("/Reviews/getReviewExcerpts/:productId : it should respond with an array, respond 200 for valid response ", async () => {
+    var productId = 1000;
+    var mockData = await SeedData.generateSeedData(50, 1);//load 10 customer,  for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviewExcerpts/${productId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toEqual(true);
+        var insertedData = response.body;
+        expect(insertedData.length).toBeLessThanOrEqual(12);
+      });
+
+  });
+
+  test("/Reviews/getReviewsByFeature/:productId : it should respond with an object, respond 200 for valid response ", async () => {
+    var productId = 1000;
+    var mockData = await SeedData.generateSeedData(10, 1);//load 10 customer,  for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviewSummary/${productId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toEqual(false);
+        expect(typeof (response.body)).toEqual('object');
+        var insertedData = response.body;
+        expect.objectContaining({
+          easeToUse: expect.any(String),
+          voiceRecognition: expect.any(String),
+          techSupport: expect.any(String),
+          valueForMoney: expect.any(String),
+          qualityOfMaterial: expect.any(String),
+          batteryLife: expect.any(String)
+        });
+      });
+
+  });
+
+
+
+});
+
```

Comment:
I think your test code looks really good. Your test are thorough and I like that you tested for valid and invalid productIds.

I don't have too many changes or comments. For your test names, you could clarify why it would have a valid response or an invalid response. 
An example would be for a valid response it needs a productId between 1000-1099.

Great job!

Reply:
Makes sense. Made a note of it.
Thanks Amos!